packages/angular-table/src/constructTableHelper.ts

```diff
@@ -0,0 +1,109 @@
+import { createColumnHelper } from '@tanstack/table-core'
+import type {
+  ColumnHelper,
+  RowData,
+  Table,
+  TableFeatures,
+  TableOptions,
+} from '@tanstack/table-core'
+import type { Signal } from '@angular/core'
+
+/**
+ * Options for creating a table helper to share common options across multiple tables
+ * columnsFeature, data, and state are excluded from this type and reserved for only the `useTable`/`createTable` functions
+ */
+export type TableHelperOptions<
+  TFeatures extends TableFeatures,
+  TData extends RowData = any,
+> = Omit<
+  TableOptions<TFeatures, NoInfer<TData>>,
+  'columns' | 'data' | 'state'
+> & {
+  _features: TFeatures
+  TData?: TData // provide a cast for the TData type
+}
+
+/**
+ * Internal type that each adapter package will build off of to create a table helper
+ */
+export type TableHelper_Core<
+  TFeatures extends TableFeatures,
+  TData extends RowData = any,
+> = {
+  columnHelper: ColumnHelper<TFeatures, TData>
+  createColumnHelper: <TData extends RowData>() => ColumnHelper<
+    TFeatures,
+    TData
+  >
+  features: TFeatures
+  options: Omit<TableOptions<TFeatures, TData>, 'columns' | 'data' | 'state'>
+  tableCreator: <TData extends RowData>(
+    tableOptions: () => Omit<
+      TableOptions<TFeatures, TData>,
+      '_features' | '_rowModels' | '_rowModelFns'
+    >,
+  ) => Table<TFeatures, TData>
+}
+
+/**
+ * Internal function to create a table helper that each adapter package will use to create their own table helper
+ */
+export function constructTableHelper<
+  TFeatures extends TableFeatures,
+  TData extends RowData = any,
+>(
+  tableCreator: (
+    tableOptions: () => TableOptions<TFeatures, TData>,
+  ) => Table<TFeatures, TData> & Signal<Table<TFeatures, TData>>,
+  tableHelperOptions: TableHelperOptions<TFeatures, TData>,
+): TableHelper_Core<TFeatures, TData> {
+  const { TData: _TData, ..._tableHelperOptions } = tableHelperOptions
+  return {
+    columnHelper: createColumnHelper<TFeatures, TData>(),
+    createColumnHelper,
+    features: tableHelperOptions._features,
+    options: _tableHelperOptions as any,
+    tableCreator: (tableOptions) =>
```

Comment:
If this is the only difference, we might explore just making some TS overloads and an option for how it needs to work for Angular in the table-core `constructTableHelper` function. Good to be here for now.